perlock-core/src/main/java/name/mitterdorfer/perlock/SinglePathWatcherFactory.java

```diff
@@ -0,0 +1,107 @@
+package name.mitterdorfer.perlock;
+
+import name.mitterdorfer.perlock.impl.util.Preconditions;
+
+import java.nio.file.Path;
+
+import java.nio.file.WatchEvent;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import static java.nio.file.StandardWatchEventKinds.*;
+
+/**
+ * Factory to create {@link PathWatcher}s that only deal with a single file and
+ * ignore change to other files within the directory
+ */
+public class SinglePathWatcherFactory {
```

Comment:
If the user calls the method `createSinglePathWatcher()` (which is intended to watch only a single file) I think the precondition `Files#isRegularFile() == true` must hold for the provided path. Otherwise they have to call another create method that is able to watch a directory, no?